### ISSUE TITLE
[dataclasses] I3Orientation::Rotate docs are misleading (Trac #1856)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1848]
The docs for I3Orientation::Rotate() state:
 Rotate a vector inside a coordinate system where z-axis==(0,0,1), x-axis==(1,0,0), y-axis=(0,1,0) to the system where z-axis==dir, x-axis==up, y-axis==right.

Taken at face value, I would expect that if I construct a vector out of scalings of the principal vectors of an I3Orientation and rotate it, I would get a vector of those scalings, e.g.:

```text
from icecube import icetray, dataclasses
from icecube.icetray import I3Units
dir = dataclasses.I3Direction(74*I3Units.degree, 22.5*I3Units.degree)
up = dataclasses.I3Direction(90*I3Units.degree, 292.5*I3Units.degree)
orie = dataclasses.I3Orientation(dir, up)
ppos = 1*orie.up + 1*orie.right + 1*orie.dir 
orie.rotate(ppos)
```

should give me `I3Position(1,1,1)`, when in fact I get `I3Position(1.65578,-0.41399,-0.294941)`. If I manually invert the quaternion inside of I3Orientation, I get the expected answer. So:

1. Is Rotate() intended to do the opposite of what it says in the docstring, i.e. rotate a vector defined in the I3Orientation's coordinate system in to the native coordinate system?
2. I have a case where I actually want the opposite operation (i.e. what it says in the docs). Should I add a `RotateIn()` method and add a `RotateOut()` method maintains the current behavior?
3. Was anyone else foolish enough to use `I3Orientation::Rotate()` in production code? I can't find any uses in IceRec at least.

Migrated from https://code.icecube.wisc.edu/ticket/1856
```json
{
    "status": "closed", 
    "changetime": "2016-09-12T19:02:59", 
    "description": "The docs for I3Orientation::Rotate() state:\n Rotate a vector inside a coordinate system where z-axis==(0,0,1), x-axis==(1,0,0), y-axis=(0,1,0) to the system where z-axis==dir, x-axis==up, y-axis==right.\n\nTaken at face value, I would expect that if I construct a vector out of scalings of the principal vectors of an I3Orientation and rotate it, I would get a vector of those scalings, e.g.:\n\n{{{\nfrom icecube import icetray, dataclasses\nfrom icecube.icetray import I3Units\ndir = dataclasses.I3Direction(74*I3Units.degree, 22.5*I3Units.degree)\nup = dataclasses.I3Direction(90*I3Units.degree, 292.5*I3Units.degree)\norie = dataclasses.I3Orientation(dir, up)\nppos = 1*orie.up + 1*orie.right + 1*orie.dir \norie.rotate(ppos)\n}}}\n\nshould give me `I3Position(1,1,1)`, when in fact I get `I3Position(1.65578,-0.41399,-0.294941)`. If I manually invert the quaternion inside of I3Orientation, I get the expected answer. So:\n\n1. Is Rotate() intended to do the opposite of what it says in the docstring, i.e. rotate a vector defined in the I3Orientation's coordinate system in to the native coordinate system?\n2. I have a case where I actually want the opposite operation (i.e. what it says in the docs). Should I add a `RotateIn()` method and add a `RotateOut()` method maintains the current behavior?\n3. Was anyone else foolish enough to use `I3Orientation::Rotate()` in production code? I can't find any uses in IceRec at least.", 
    "reporter": "jvansanten", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1473706979023252", 
    "component": "combo core", 
    "summary": "[dataclasses] I3Orientation::Rotate docs are misleading", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-09-12T12:40:34", 
    "milestone": "", 
    "owner": "claudio.kopper", 
    "type": "rumor / allegation"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160912T15:11:45: claudio.kopper commented:

> Maybe the safest way would be to create RotateIn() and RotateOut() as you suggested and remove the original function. That would make existing code fail to compile but I don't think the function is actually used anywhere. (And if it is we can either change that code or think about restoring it).

**Comment by icecube-trac** [comment]
Trac update at 20160912T19:02:59: 
* jvansanten commented:

> In changeset 149975:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149975"
> Remove I3Orientation::Rotate(), and replace it with the pair of
> complementary methods RotateIn() and RotateOut(). RotateOut() performs
> the operation that Rotate() implemented, while RotateIn() does what
> the docs said Rotate() did. Along the way, add a test that defines the
> intended behavior of the rotation. Fixes #1848.
> ```


* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "fixed"
